File: Otimizados/policripto.c
#include <stdio.h>
#include <string.h>

int index_of(char str[], char chr) {
    int i;
    for (i = 0; i < 26; i++)
        if (str[i] == chr)
            return i;
    return -1;
}

int main(void) {
    char entrada[201], nums_[401], sinal = '\0', alfa[] = "abcdefghijklmnopqrstuvwxyz";
    int nums[400], i, indexNum = 0, i_ = 0;

    scanf("%s", entrada);
    getchar();

    scanf("%s", nums_);
    getchar();

    for(i = 0; i < strlen(nums_); i++) {
        char num = nums_[i];
        if (num == '-' || num == '+') sinal = num;
        else {
            int valor = num - '0';
            if (sinal == '-') valor *= -1;
            nums[indexNum] = valor;
            indexNum++;
        }
    }

    for (i = 0; i < strlen(entrada); i++) {
        char letra = entrada[i];
        if (letra == 'w') {
            printf(" ");
            continue;
        }
        int posicao = index_of(alfa, letra);
        posicao += nums[i_];
        i_++;
        if (posicao > 25) posicao -= 26;
        if (posicao < 0) posicao += 26;
        printf("%c", alfa[posicao]);
    }
    printf("\n");
    return 0;
}

File: policripto.py
entrada = input()
nums_ = [x for x in input()]
nums = []
sinal = ""

for num in nums_:
    if num == "-" or num == "+": sinal = num
    else:
        valor = int(num)
        if sinal == "-": valor = valor * -1
        nums.append(valor)

alfa = "abcdefghijklmnopqrstuvwxyz"    
i = 0
for letra in entrada:
    if letra == 'w': 
        print(' ', end='')
        continue
    posicao = alfa.index(letra)
    posicao = posicao + nums[i]
    if (posicao > 25):
        posicao = posicao - 26
    print(alfa[posicao], end='')
    i = i + 1
print()